internal/controller/inference_pool.go

```diff
@@ -0,0 +1,409 @@
+func (c *InferencePoolController) getReferencedGateways(ctx context.Context, inferencePool *gwaiev1a2.InferencePool) (map[string]*gwapiv1.Gateway, error) {
+	// Find all Gateways across all namespaces.
+	var gateways gwapiv1.GatewayList
+	if err := c.client.List(ctx, &gateways); err != nil {
+		return nil, fmt.Errorf("failed to list Gateways: %w", err)
+	}
+
+	referencedGateways := make(map[string]*gwapiv1.Gateway)
+
+	// Check each Gateway to see if it references this InferencePool through routes.
+	for _, gw := range gateways.Items {
+		if c.gatewayReferencesInferencePool(ctx, &gw, inferencePool.Name, inferencePool.Namespace) {
+			gatewayKey := fmt.Sprintf("%s/%s", gw.Namespace, gw.Name)
+			referencedGateways[gatewayKey] = &gw
+		}
+	}
+
+	return referencedGateways, nil
+}
+
+// validateExtensionReference checks if the ExtensionReference service exists.
+func (c *InferencePoolController) validateExtensionReference(ctx context.Context, inferencePool *gwaiev1a2.InferencePool) error {
+	// Check if ExtensionRef is specified.
+	if inferencePool.Spec.ExtensionRef == nil {
+		return nil // No extension reference to validate.
+	}
+
+	// Get the service name from ExtensionReference.
+	serviceName := inferencePool.Spec.ExtensionRef.Name
+	if serviceName == "" {
+		return fmt.Errorf("ExtensionReference name is empty")
+	}
+
+	// Check if the service exists.
+	var service corev1.Service
+	if err := c.client.Get(ctx, client.ObjectKey{
+		Name:      string(serviceName),
+		Namespace: inferencePool.Namespace,
+	}, &service); err != nil {
+		if client.IgnoreNotFound(err) == nil {
+			// Service not found - this is the error case we want to handle.
+			return fmt.Errorf("ExtensionReference service %s not found in namespace %s", serviceName, inferencePool.Namespace)
+		}
+		// Other error occurred.
+		return fmt.Errorf("failed to get ExtensionReference service %s: %w", serviceName, err)
+	}
+
+	// Service exists - validation passed.
+	return nil
+}
+
+// gatewayReferencesInferencePool checks if a Gateway references the given InferencePool through any routes.
+func (c *InferencePoolController) gatewayReferencesInferencePool(ctx context.Context, gateway *gwapiv1.Gateway, inferencePoolName string, inferencePoolNamespace string) bool {
+	// Check AIGatewayRoutes in the same namespace as the InferencePool that reference this Gateway.
+	var aiGatewayRoutes aigv1a1.AIGatewayRouteList
+	if err := c.client.List(ctx, &aiGatewayRoutes, client.InNamespace(inferencePoolNamespace)); err != nil {
+		c.logger.Error(err, "failed to list AIGatewayRoutes", "gateway", gateway.Name, "namespace", inferencePoolNamespace)
+		return false
+	}
+
+	for _, route := range aiGatewayRoutes.Items {
+		// Check if this route references the Gateway.
+		if c.routeReferencesGateway(route.Spec.ParentRefs, gateway.Name, gateway.Namespace) {
+			// Check if this route references the InferencePool.
+			if c.routeReferencesInferencePool(&route, inferencePoolName) {
+				return true
+			}
+		}
+	}
```

Comment:
You are escaping the route unnecessarily

```suggestion
	for i := range aiGatewayRoutes.Items {
        route := &aiGatewayRoutes.Items[i]
		// Check if this route references the Gateway.
		if c.routeReferencesGateway(route.Spec.ParentRefs, gateway.Name, gateway.Namespace) {
			// Check if this route references the InferencePool.
			if c.routeReferencesInferencePool(route, inferencePoolName) {
				return true
			}
		}
	}
```